pkg/events/redis/store.go

```diff
@@ -399,7 +400,7 @@ func (ps *PubSubStore) Publish(evt events.Event) {
 				Values:       streamValues,
 			})
 			tx.Expire(ps.ctx, eventStream, ps.entityHistoryTTL)
-			if devID := id.GetDeviceIds(); devID != nil {
+			if devID := id.GetDeviceIds(); devID != nil && definition != nil && definition.PropagateToParent() {
```

Comment:
In which situation wouldn't a `nil` definition be an implementation error ? (i.e. publishing something with an unregistered name). 

I can understand how between versions you may not be able to have definition while _receiving an event_, but how would that happen on `Publish` ?